langchain/embeddings/huggingface.py

```diff
@@ -1,17 +1,23 @@
 """Wrapper around HuggingFace embedding models."""
 from typing import Any, List
+from enum import Enum
 
 from pydantic import BaseModel, Extra
 
 from langchain.embeddings.base import Embeddings
 
 DEFAULT_MODEL_NAME = "sentence-transformers/all-mpnet-base-v2"
+DEFAULT_INSTRUCTION = "Represent the following text:"
```

Comment:
Reading the Instructor paper - they support asymmetric instructions (ie. the instruction for embedding and the instruction for retrieval are different). From their repo, "Represent the Wikipedia document for retrieval: " is used for the original embedding and "Represent the Wikipedia question for retrieving supporting documents: " is used when constructing the query embedding. Would be good to support this.

Reply:
I agree, however adding in the ability to do this makes the code a little wonky. Would love to get this merged first as a V1 and then add the asymmetric instructions component later. Alternatively, if you can figure out a good way to do this feel free to add it to this PR. Thoughts?

Reply:
I think the idea of asymmetric prompts for an embed/query class like this feels somewhat generic - so my instinct would be just to have two parameters to the class `embed_instuction` and `query_instruction` instead of just `instruction`. They can both default to `DEFAULT_INSTRUCTION` but then the embed and query methods can just reference the correct one. I am not a good judge if this is 'wonky' though. 

I agree that this would be ok to merge as is for v1 though - so maybe time for @hwchase17 to have a look again